lib/ansible/modules/packaging/os/portage.py

```diff
@@ -490,9 +511,16 @@ def main():
             keepgoing=dict(default=False, type='bool'),
             jobs=dict(default=None, type='int'),
             loadavg=dict(default=None, type='float'),
+            quietbuild=dict(default=False, type='bool'),
```

Comment:
How about `default=None`? Otherwise it might override `EMERGE_DEFAULT_OPTS` values, which would be weird.

Reply:
I was just being consistent. There are only 3 options that are set to None, all others False. Once merged I plan to switch some of these defaults to True. I feel like ansible should always default to quiet across the board, unless overridden. 

But you make a good point. It maybe worth dropping all False to None. I think its that way based on [the code](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/portage.py#L324). Not familiar with conditionals, but I guess None is the same as False? If that is the case then its moot. Either way I believe it only adds these if set to True. If set to False they do not show up. Or should not per how I read the code.

Reply:
No, `None is not False` (yes, it's valid Python statement bwt :smiley:)

Basically, `None` *value* **is casted to False**. *But* `if var` check is using a built-in pythonic protocol for casting bool value out of expression given.
So it does `bool(val)`, which internally does `var.__bool__()` call. And then it uses whatever this method returns for the `if` clause.

`NoneType` (`type(None)`) is always falsy, `[]` is falsy, `0` is falsy, `{}` is falsy. `1` is truthy, `[0]` is truthy, `{'k': 'v'}` is truly.
The end-user can implement whatever logic they want in their classes for casting their instances into boolean.

That is why it's important to check for `val is None` or `val is not None`, sometimes you'll even want `val is False`. `None` is often used for default arg value for functions/methods, which then do `is None` check to ensure the user didn't override them (cause you cannot put mutable defaults in function definitions).

Reply:
Ok based on that it seems the conditional to add any flag requires it to be set to True, per the [code](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/portage.py#L324).
```python
    for flag, arg in emerge_flags.items():
        if p[flag]:
             args.append(arg)
```

If I set it to None it would not be handled correctly. Or maybe ignored since its not True. Pretty sure that will only evaluate to true, if set to True.

Reply:
Yes, it will append flags with `True`, `1`, `1.0`, `y`, `n` (any string) values. And will skip `0`, `0.0`, `None`, `False`, `[]`, `{}`.

Reply:
So having None vs False is moot for that stuff. Seems only remaining change is the version. I will correct and break CI

Reply:
Yes. We'll probably need to consider redesigning defaults separately. Out of scope now.

Reply:
The module could likely use a few improvements. I am open to such. Not sure how many are using the module. It does not seem very popular, or its meeting alls needs/expectations no clue. Then again not sure many are interested in Gentoo these days...